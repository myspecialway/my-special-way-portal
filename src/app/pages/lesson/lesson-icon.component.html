<div class="lesson-icon-container">
    abcd
</div>