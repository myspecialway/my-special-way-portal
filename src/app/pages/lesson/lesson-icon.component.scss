.lesson-icon-container {
  width: 80px;
  height: 80px;
  border-radius: 2px;
  background-color: #ffffff;
  text-align: center;
  vertical-align: middle;
  float: left;
  margin: 5px;
}
.lesson-icon-selected {
  border-radius: 2px;
  border: solid 2px #8b66a7;
}
