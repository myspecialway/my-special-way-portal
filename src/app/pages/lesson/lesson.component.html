<div data-test-id="lessons-page" class="main-content">
  <div class="container-fluid">
    <div class="header-menu-container msw-table-tools-header">
      <div class="header-table-count">
        <div>
          <span class="header-table-count-label">שיעורים</span>
          <span class="header-table-count-number">({{dataSource.data.length}})</span>
        </div>
      </div>
      <div class="example-header msw-table-tools-header">
        <button mat-raised-button color="primary" (click)="addNewLesson()">
          <i class="material-icons">
            add
          </i>
          הוסף שיעור חדש
        </button>

      </div>
    </div>
    <div class="row msw-table-container">
        <div class="col-lg-12 col-md-12 col-sm-12">
            <mat-table #table [dataSource]="dataSource" class="msw-table"
            matSort matSortActive="title" matSortDisableClear matSortDirection="asc">

                <!-- Name Column -->
                <ng-container matColumnDef="title">
                  <mat-header-cell *matHeaderCellDef mat-sort-header>שם שיעור</mat-header-cell>
                  <mat-cell *matCellDef="let row"> {{row.title}}</mat-cell>
                </ng-container>

                <!-- Icon Column -->
                <ng-container matColumnDef="icon">
                  <mat-header-cell *matHeaderCellDef mat-sort-header>סימול</mat-header-cell>
                  <mat-cell *matCellDef="let row">{{row.icon}}</mat-cell>
                </ng-container>

                <ng-container matColumnDef="deleteLesson">
                    <mat-header-cell *matHeaderCellDef></mat-header-cell>
                    <mat-cell *matCellDef="let row">
                      <button class="delete-row-button" mat-icon-button color="primary" (click)="deleteLesson(row._id)"><i class="material-icons"> delete </i></button>
                    </mat-cell>
                </ng-container>

              <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
              <mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row>
            </mat-table>
          </div>
    </div>
  </div>
</div>
