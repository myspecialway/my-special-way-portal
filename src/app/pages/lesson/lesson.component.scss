@import '../../../assets/scss/msw-table';
img {
  width: 65.1px;
  height: 59.2px;
}
