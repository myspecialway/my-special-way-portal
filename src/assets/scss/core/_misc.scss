body {
  background-color: #fafafa;
  color: $black-color;
  font-weight: 300;
}

legend {
  border-bottom: 0;
}

.serif-font {
  font-family: $font-family-serif;
}

// Prevent highlight on mobile
* {
  -webkit-tap-highlight-color: rgba(255, 255, 255, 0);
  -webkit-tap-highlight-color: transparent;
  &:focus {
    outline: 0;
  }
}

a {
  color: $link-color;
  &:hover,
  &:focus {
    color: darken($link-color, 5%);
    text-decoration: none;
  }

  &.text-info {
    &:hover,
    &:focus {
      color: darken($brand-info, 5%);
    }
  }

  & .material-icons {
    vertical-align: middle;
  }
}

.form-check,
label {
  font-size: 14px;
  line-height: 1.42857;
  color: $checkboxes-text-color;
  font-weight: 400;
}

/*           Animations              */
.animation-transition-general {
  @include transition-all($general-transition-time, $transition-linear);
}

.animation-transition-slow {
  @include transition-all($slow-transition-time, $transition-linear);
}

.animation-transition-fast {
  @include transition-all($fast-transition-time, $transition-ease);
}

.caret,
.sidebar a {
  @include transition-all($fast-transition-time, $transition-ease-in);
}
#map {
  position: relative;
  z-index: 2;
  width: 100%;
  height: calc(100vh - 70px);
  margin-top: 70px;
}

@font-face {
  font-family: 'pass';
  font-style: normal;
  font-weight: 400;
  src: url(data:application/font-woff;charset=utf-8;base64,d09GRgABAAAAAATsAA8AAAAAB2QAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABGRlRNAAABWAAAABwAAAAcg9+z70dERUYAAAF0AAAAHAAAAB4AJwANT1MvMgAAAZAAAAA/AAAAYH7AkBhjbWFwAAAB0AAAAFkAAAFqZowMx2N2dCAAAAIsAAAABAAAAAQAIgKIZ2FzcAAAAjAAAAAIAAAACAAAABBnbHlmAAACOAAAALkAAAE0MwNYJ2hlYWQAAAL0AAAAMAAAADYPA2KgaGhlYQAAAyQAAAAeAAAAJAU+ATJobXR4AAADRAAAABwAAAAcCPoA6mxvY2EAAANgAAAAEAAAABAA5gFMbWF4cAAAA3AAAAAaAAAAIAAKAE9uYW1lAAADjAAAARYAAAIgB4hZ03Bvc3QAAASkAAAAPgAAAE5Ojr8ld2ViZgAABOQAAAAGAAAABuK7WtIAAAABAAAAANXulPUAAAAA1viLwQAAAADW+JM4eNpjYGRgYOABYjEgZmJgBEI2IGYB8xgAA+AANXjaY2BifMg4gYGVgYVBAwOeYEAFjMgcp8yiFAYHBl7VP8wx/94wpDDHMIoo2DP8B8kx2TLHACkFBkYA8/IL3QB42mNgYGBmgGAZBkYGEEgB8hjBfBYGDyDNx8DBwMTABmTxMigoKKmeV/3z/z9YJTKf8f/X/4/vP7pldosLag4SYATqhgkyMgEJJnQFECcMOGChndEAfOwRuAAAAAAiAogAAQAB//8AD3jaY2BiUGJgYDRiWsXAzMDOoLeRkUHfZhM7C8Nbo41srHdsNjEzAZkMG5lBwqwg4U3sbIx/bDYxgsSNBRUF1Y0FlZUYBd6dOcO06m+YElMa0DiGJIZUxjuM9xjkGRhU2djZlJXU1UDQ1MTcDASNjcTFQFBUBGjYEkkVMJCU4gcCKRTeHCk+fn4+KSllsJiUJEhMUgrMUQbZk8bgz/iA8SRR9qzAY087FjEYD2QPDDAzMFgyAwC39TCRAAAAeNpjYGRgYADid/fqneL5bb4yyLMwgMC1H90HIfRkCxDN+IBpFZDiYGAC8QBbSwuceNpjYGRgYI7594aBgcmOAQgYHzAwMqACdgBbWQN0AAABdgAiAAAAAAAAAAABFAAAAj4AYgI+AGYB9AAAAAAAKgAqACoAKgBeAJIAmnjaY2BkYGBgZ1BgYGIAAUYGBNADEQAFQQBaAAB42o2PwUrDQBCGvzVV9GAQDx485exBY1CU3PQgVgIFI9prlVqDwcZNC/oSPoKP4HNUfQLfxYN/NytCe5GwO9/88+/MBAh5I8C0VoAtnYYNa8oaXpAn9RxIP/XcIqLreZENnjwvyfPieVVdXj2H7DHxPJH/2/M7sVn3/MGyOfb8SWjOGv4K2DRdctpkmtqhos+D6ISh4kiUUXDj1Fr3Bc/Oc0vPqec6A8aUyu1cdTaPZvyXyqz6Fm5axC7bxHOv/r/dnbSRXCk7+mpVrOqVtFqdp3NKxaHUgeod9cm40rtrzfrt2OyQa8fppCO9tk7d1x0rpiQcuDuRkjjtkHt16ctbuf/radZY52/PnEcphXpZOcofiEZNcQAAeNpjYGIAg///GBgZsAF2BgZGJkZmBmaGdkYWRla29JzKggxD9tK8TAMDAxc2D0MLU2NjENfI1M0ZACUXCrsAAAABWtLiugAA)
    format('woff');
}

input[type='password'] {
  font-family: 'pass', 'Roboto', Helvetica, Arial, sans-serif;
  font-size: 18px;
  &::-webkit-input-placeholder {
    transform: scale(0.77);
    transform-origin: 0 50%;
  }
  &::-moz-placeholder {
    font-size: 14px;
    opacity: 1;
  }
  &:-ms-input-placeholder {
    font-size: 14px;
    font-family: 'Roboto', Helvetica, Arial, sans-serif;
  }
}
