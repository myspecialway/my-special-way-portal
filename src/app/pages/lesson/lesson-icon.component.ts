import { Component, OnInit, Input } from '@angular/core';

@Component({
  selector: 'app-lesson-icon',
  templateUrl: './lesson-icon.component.html',
  styleUrls: ['./lesson-icon.component.scss'],
})
export class LessonIconComponent implements OnInit {
  @Input()
  public icon = '';
  @Input()
  public data: { icon: string };

  public ngOnInit(): void {}
}
