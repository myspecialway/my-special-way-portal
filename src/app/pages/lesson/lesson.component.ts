import {MatTableDataSource, MatSort } from '@angular/material';
import { Component, OnInit, ViewChild } from '@angular/core';
import { Lesson } from '../../models/lesson.model';
import { LessonService } from '../../services/lesson/lesson.graphql.service';

@Component({
  selector: 'app-lesson',
  templateUrl: './lesson.component.html',
  styleUrls: ['./lesson.component.scss'],
})
export class LessonComponent implements OnInit {

  displayedColumns = ['title', 'icon', 'deleteLesson'];
  dataSource = new MatTableDataSource<Lesson>();

  @ViewChild(MatSort) sort: MatSort;

  constructor(
    private lessonService: LessonService,

  ) { }

  async ngOnInit() {
     const lessons = await this.lessonService.getLessons();
     this.dataSource.data = [...lessons];
  }

  addNewLesson() {
  }

  deleteLesson(_id: number) {

  }
}
