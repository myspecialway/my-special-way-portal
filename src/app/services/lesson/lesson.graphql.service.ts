import { Injectable } from '@angular/core';
import { Apollo } from 'apollo-angular';
import gql from 'graphql-tag';
import { LessonQuery } from '../../models/lesson.model';

@Injectable()
export class LessonService {
  constructor(private apollo: Apollo) {}

  getLessons() {
    return this.apollo
      .query<LessonQuery>({
        query: gql`
          {
            lessons {
              _id
              title
              icon
            }
          }
        `,
      })
      .toPromise().then((res) => res.data.lessons);
  }
}
