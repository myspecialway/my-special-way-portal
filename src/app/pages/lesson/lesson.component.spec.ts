import { TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { LessonComponent } from './lesson.component';
import { LessonService } from '../../services/lesson/lesson.graphql.service';
import { Lesson } from '../../models/lesson.model';
import { MatHeaderRow, MatRowDef, MatHeaderRowDef, MatSort, MatPaginator, MatDialog } from '@angular/material';

import { Observable } from 'rxjs-compat';
import { ClassService } from '../class/services/class.graphql.service';

describe('lesson component', () => {
  let lessonServiceMock: Partial<LessonService>;
  let classServiceMock: Partial<ClassService>;
  let lessonTestData: Lesson[] = [];
  let lessonDialogMock: Partial<MatDialog>;
  let lessonDialogMockValue: boolean;
  beforeAll(async () => {
    lessonTestData = [
      {
        _id: '1',
        title: 'Sport',
        icon: 's',
      },
      {
        _id: '2',
        title: 'Art',
        icon: 'a',
      },
    ];
  });

  beforeEach(async () => {
    lessonServiceMock = {
      getLessons: jest.fn(),
      getAllLessons: jest.fn(),
      delete: jest.fn(),
    };
    classServiceMock = {
      getAllClasses: jest.fn().mockReturnValue(Observable.of([])),
    };
    lessonDialogMockValue = true;
    lessonDialogMock = {
      open: jest.fn().mockReturnValue({
        afterClosed: jest.fn().mockImplementation(() => Observable.of(lessonDialogMockValue)),
      }),
    };
    TestBed.configureTestingModule({
      imports: [],
      declarations: [LessonComponent, MatHeaderRow, MatRowDef, MatHeaderRowDef, MatSort, MatPaginator],
      providers: [
        { provide: LessonService, useValue: lessonServiceMock },
        { provide: MatDialog, useValue: lessonDialogMock },
        { provide: ClassService, useValue: classServiceMock },
      ],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  it('should render component as described in snapshot', () => {
    const fixture = TestBed.createComponent(LessonComponent);
    expect(fixture).toMatchSnapshot();
  });

  it('should load correct number of lesson ', async () => {
    (lessonServiceMock.getLessons as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(lessonTestData);
    });
    (lessonServiceMock.getAllLessons as jest.Mock).mockImplementationOnce(() => {
      return Observable.of(lessonTestData);
    });
    const fixture = TestBed.createComponent(LessonComponent);
    fixture.detectChanges();
    await fixture.whenRenderingDone();
    expect(fixture.componentInstance.dataSource.data.length).toEqual(2);
  });

  it('should load zero lesson in case of promise reject', async () => {
    (lessonServiceMock.getLessons as jest.Mock).mockImplementationOnce(() => {
      return Promise.reject();
    });
    const fixture = TestBed.createComponent(LessonComponent);
    fixture.detectChanges();
    await fixture.whenRenderingDone();
    expect(fixture.componentInstance.dataSource.data.length).toEqual(0);
  });

  it('should should open delete dialoge when calling delete component ', async () => {
    (lessonServiceMock.getLessons as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(lessonTestData);
    });
    (lessonServiceMock.delete as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(1);
    });
    const fixture = TestBed.createComponent(LessonComponent);
    await fixture.componentInstance.ngOnInit(); // this triggers the subCleaner instantiator.
    await fixture.componentInstance.deleteLesson('1', 'anyTitle');
    fixture.detectChanges();
    await fixture.whenRenderingDone();
    expect(lessonDialogMock.open).toHaveBeenCalled();
    //expect(fixture.componentInstance.dataSource.data.length).toEqual(1);
  });
  it('should call delete lesson when user confirms ', async () => {
    (lessonServiceMock.getLessons as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(lessonTestData);
    });
    (lessonServiceMock.delete as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(1);
    });
    lessonDialogMockValue = true;
    const fixture = TestBed.createComponent(LessonComponent);
    await fixture.componentInstance.ngOnInit(); // this triggers the subCleaner instantiator.
    await fixture.componentInstance.deleteLesson('1', 'anyTitle');
    fixture.detectChanges();
    await fixture.whenRenderingDone();
    expect(lessonServiceMock.delete).toHaveBeenCalledWith('1');
  });

  it('should not call delete lesson when user cancels ', async () => {
    (lessonServiceMock.getLessons as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(lessonTestData);
    });
    (lessonServiceMock.delete as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(1);
    });
    lessonDialogMockValue = false;

    const fixture = TestBed.createComponent(LessonComponent);
    await fixture.componentInstance.ngOnInit(); // this triggers the subCleaner instantiator.
    await fixture.componentInstance.deleteLesson('1', 'anyTitle');
    fixture.detectChanges();
    await fixture.whenRenderingDone();
    expect(lessonServiceMock.delete).not.toHaveBeenCalled();
  });
});
