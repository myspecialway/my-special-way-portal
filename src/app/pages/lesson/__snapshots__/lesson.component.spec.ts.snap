// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lesson component should render component as described in snapshot 1`] = `
<app-lesson
  dataSource={[Function MatTableDataSource]}
  displayedColumns={[Function Array]}
  lessonService={[Function Object]}
  sort={[Function MatSort]}
>
  <div
    class="main-content"
  >
    <div
      class="container-fluid"
    >
      <div
        class="header-menu-container msw-table-tools-header"
      >
        <div>
           שיעורים (
        </div>
        <div
          class="example-header"
        >
          <button
            color="primary"
            mat-raised-button=""
          >
            הוסף שיעור חדש
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-12 col-md-12 col-sm-12"
        >
          <mat-table
            class="msw-table"
            matsort=""
            matsortactive="title"
            matsortdirection="asc"
            matsortdisableclear=""
          >
            
            
            
            
            
            
            
            --&gt; 
            
            
            
            
          </mat-table>
        </div>
      </div>
    </div>
  </div>
</app-lesson>
`;
